[BLAS, cuBLAS backend, hipSYCL] Add official hipsycl cuBLAS support

sbalint98 opened the pull request:
As discussed under PR #144 hipSYCL rocBLAS support comes with automatic support for cuBLAS as well. This PR adds this information to the README. 

Furthermore, this PR differentiates the FindcuBLAS.cmake file between hipSYCL and Intel LLVM. In the case of hipSYCL, there is no need to have the `OPENCL_PATH` set. Moreover, the `CUDA_CUDA_LIBRARY` environment variable is also not strictly necessary since finding the libcuda.so is handled by hipSYCL.

[onemkl-cuda-hipsycl-test.txt](https://github.com/oneapi-src/oneMKL/files/8076543/onemkl-cuda-hipsycl-test.txt)

sbalint98 committed:
Add hipSYCL cuBLAS support to the README

sbalint98 committed:
Update find cuBLAS cmake for hipSYCL

mmeterel commented:
@sbalint98 As you are aware, there were several changes since the time this PR is opened. Could you please rebase and check if the tests are all good? Also, I think, we can enable cuRAND backend with hipSYCL with this PR as well. What do you think?

Could you please let me know when everything is ready?

sbalint98 commented:
Hi @mmeterel thank you for reminding me. I'll have a look and prepare this PR until Monday. I'll let you know if it is ready for review

sbalint98 commented:
Sorry for the delay. I have rebased the branch and updated the readme. All tests are passing on my sid
[cublas_tests_hipsycl.txt](https://github.com/oneapi-src/oneMKL/files/8846920/cublas_tests_hipsycl.txt)
e

mmeterel commented:
> Sorry for the delay. I have rebased the branch and updated the readme. All tests are passing on my sid [cublas_tests_hipsycl.txt](https://github.com/oneapi-src/oneMKL/files/8846920/cublas_tests_hipsycl.txt) e

@sbalint98 Thanks. Do you plan to add cuRAND support in this PR as well?

sbalint98 commented:
@nilsfriess  and @normallytangent are working on the cuRAND support. I believe for them it might be easier to open a new PR. Would this be acceptable for you?

mmeterel commented:
> @nilsfriess and @normallytangent are working on the cuRAND support. I believe for them it might be easier to open a new PR. Would this be acceptable for you?

Sure, that works if it will be easier on your side.

mmeterel commented:
@sbalint98 
Thanks for the PR. I tested it locally (had to rebase hipSYCL and build it with LLVM-13) and it works smoothly. :)

mmeterel commented:
@mkrainiuk Could you please take a look at this?

mkrainiuk commented on cmake/FindcuBLAS.cmake:
```diff
@@ -32,20 +32,36 @@ add_compile_definitions(CUDA_NO_HALF)
+          CUDA_LIBRARIES
+          CUDA_CUDART_LIBRARY
+    )
+    set_target_properties(ONEMKL::cuBLAS::cuBLAS PROPERTIES
```
Is there any reasons we cannot use one `set_target_properties`? I see the only difference: `OPENCL_INCLUDE_DIR` that can be added with inline check, something like "$<$<BOOL:${USE_ADD_SYCL_TO_TARGET_INTEGRATION}>: ${OPENCL_INCLUDE_DIR}>"

mmeterel replied:
Not sure...but i also see differences between CUDA_CUDART_LIBRARY vs CUDA_CUDA_LIBRARY

mkrainiuk reviewed (approved):
Thank you for the PR! Changes look good to me. I have one minor comment.
